lib/pure/lambdas.nim

```diff
@@ -0,0 +1,111 @@
+#[
+D20190811T003919
+]#
+
+type aliassym* {.magic: AliasSym.} # RENAME
+
+proc alias2*[T](x: T): aliassym {.magic: "Alias2", compileTime.} # TODO: RENAME
+  # Declares `name` as alias of `expr`, which must resolve to a symbol.
+  # Works with any symbol, e.g. iterator, template, macro, module, proc etc.
+
+import std/macros
+
+template lambdaIter*(a: untyped): untyped =
+  block:
+    template lambdaIterImpl(): untyped = a
+    alias2(lambdaIterImpl)
+
+template lambdaIt*(a: untyped): untyped =
+  block:
+    template lambdaItImpl(a0): untyped =
+      block:
+        var it {.inject.} = a0
+        a
+    alias2(lambdaItImpl)
+
+proc replaceSym(body: NimNode, sym: NimNode): NimNode =
+  ## xxx expose in macros.nim
+  case body.kind
+  of nnkSym:
+    if body == sym:
+      return ident(sym.strVal)
+  else:
+    for i in 0..<len(body):
+      body[i] = replaceSym(body[i], sym)
+  return body
+
+macro `~>`*(lhs, rhs: untyped): untyped =
```

Comment:
The implementation of this could use some comments explaining what the goal of the code is, ie, "this goal of this macro is to generate a template that ... ")

Reply:
done, added doc comments + runnableExamples